regulations/tasks.py

```diff
@@ -19,63 +19,75 @@
-
-            # Add other submitted fields
-            fields.extend([
-                (name, value)
-                for name, value in six.iteritems(body)
-                if name != 'general_comment'
-            ])
-            fields.extend(attachments)
-
-            data = MultipartEncoder(fields)
-            response = requests.post(
-                settings.REGS_GOV_API_URL,
-                data=data,
-                headers={
-                    'Content-Type': data.content_type,
-                    'X-Api-Key': settings.REGS_GOV_API_KEY,
-                }
-            )
-            if response.status_code != requests.codes.created:
-                logger.warn("Post to regulations.gov failed: %s %s",
-                            response.status_code, response.text)
-                raise self.retry()
-            logger.info(response.text)
-            return response.json()
-    except (ClientError, RequestException) as exc:
-        logger.exception(exc)
-        raise self.retry(exc=exc)
+        html = json_to_html(body)
+        files = extract_files(body['assembled_comment'])
+        try:
+            with html_to_pdf(html) as comment, \
+                    build_attachments(files) as attachments:
+                fields = [
+                    ('comment_on', settings.COMMENT_DOCUMENT_ID),
+                    # TODO: Ensure this name is unique
+                    ('uploadedFile', ('comment.pdf', comment)),
+                    ('general_comment', 'See attached comment.pdf'),
+                ]
+
+                # Add other submitted fields
+                fields.extend([
+                    (name, value)
+                    for name, value in six.iteritems(body)
+                    if name != 'assembled_comment'
+                ])
+                fields.extend(attachments)
+
+                data = MultipartEncoder(fields)
+                response = requests.post(
+                    settings.REGS_GOV_API_URL,
+                    data=data,
+                    headers={
+                        'Content-Type': data.content_type,
+                        'X-Api-Key': settings.REGS_GOV_API_KEY,
+                    }
+                )
+                if response.status_code != requests.codes.created:
+                    logger.warn("Post to regulations.gov failed: %s %s",
+                                response.status_code, response.text)
+                    raise self.retry()
+                logger.info(response.text)
+                return response.json()
+        except (RequestException, ClientError):
+            logger.exception("Unable to submit comment")
+            raise self.retry()
+    except MaxRetriesExceededError:
+        message = "Exceeded retries, saving failed submission"
+        logger.error(message)
+        failed_submission = FailedCommentSubmission(body=json.dumps(body))
+        failed_submission.save()
+        return {'message': message, 'trackingNumber': None}
```

Comment:
Shouldn't hold this PR back, but the logic in this function is becoming relatively complex which is difficult for both reasoning and testing. Can it be broken up?

Also, very minor, but if you're not going to use the `failed_submission` object, would it make sense to do
```python
FailedCommentSubmission.objects.create(body=json.dumps(body))
```
?